src/main/java/com/minecolonies/coremod/event/EventHandler.java

```diff
@@ -314,6 +322,30 @@ else if ("pmardle".equalsIgnoreCase(event.getEntityPlayer().getName())
             {
                 return;
             }
+            if (world.getBlockState(event.getPos()).getBlock().isBed(world.getBlockState(event.getPos()), world, event.getPos(), player))
+            {
+
+                final IColony colony = ColonyManager.getIColony(world, event.getPos());
+                //Checks to see if player tries to sleep in a bed belonging to a Citizen, ancels the event, and Notifies Player that bed is occuppied
+                if (colony != null)
+                {
+                    final List<CitizenData> citizenList = ColonyManager.getClosestColony(world, event.getPos()).getCitizenManager().getCitizens();
```

Comment:
why don't you get a real colony above already, there is a ColonyManager.getColonyByPosFromWorld() then you don't have to get the colony twice.

Reply:
I don't see that function anywhere. I have GetClosestColony, and GetClosestIColony